Tickr/Tickr/SearchTableViewController.swift

```diff
@@ -0,0 +1,141 @@
+                self.searchResults = stockInfoArray
+                self.tableView.reloadData()
+            })
+        }
+    }
+}
+
+//MARK: - TableView Delegate and Datasource
+extension SearchTableViewController: UITableViewDelegate, UITableViewDataSource {
+    
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return searchResults.count
+    }
+    
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithIdentifier("searchResultCell") as! SearchResultCell
+        cell.parentVC = self
+        cell.configureCellWithSearchResult(searchResults[indexPath.row])
+        
+        return cell
+    }
+    
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        let cell = tableView.cellForRowAtIndexPath(indexPath) as! SearchResultCell
+        cell.addButtonWasPressed(nil)
+    }
+}
+
+
+//MARK: - SearchBar Delegate
+extension SearchTableViewController: UISearchBarDelegate {
+    
+    func searchBar(searchBar: UISearchBar, textDidChange searchText: String) {
+        //If there is user input, suggest companies that match their query
+        let length = searchText.characters.count
+        if length > 0 {
+            searchYahooFinanceWithString(searchText)
+        } else {
+            searchResults.removeAll()
+            tableView.reloadData()
+        }
+    }
+    
+    func searchBarShouldBeginEditing(searchBar: UISearchBar) -> Bool {
+        searchBar.showsCancelButton = true
+        tableView.reloadData()
+        return true
+    }
+    
+    func searchBarShouldEndEditing(searchBar: UISearchBar) -> Bool {
+        searchBar.showsCancelButton = false
+        tableView.reloadData()
+        return true
+    }
+    
+    //dismiss search VC when cancel button is selected
+    func searchBarCancelButtonClicked(searchBar: UISearchBar) {
+        searchBar.resignFirstResponder()
+        searchBar.text = ""
+        self.dismissViewControllerAnimated(false, completion: nil)
+    }
+}
+
+
+//MARK: - SearchResultCell Class
+//A simple UITableViewCell for displaying search results
+class SearchResultCell: UITableViewCell {
+    //Class Variables
+    var parentVC: SearchTableViewController!
+    
+    //Outlets
+    @IBOutlet weak var symbolLabel: UILabel!
+    @IBOutlet weak var nameLabel: UILabel!
+    
+    //Actions
+    @IBAction func addButtonWasPressed(sender: UIButton?) {
+        //create a new stock instance
+        let newStock = Stock(
+            name: nameLabel.text!,
+            symbol: symbolLabel.text!,
+            price: 0.0,
+            netChange: 0.0,
+            netChangeInPercentage: 0.0
+        )
+        
+        //Try to add this stock to the watchlist.
+        WatchListManager.sharedInstance.addStockToWatchList(newStock, completion: { (stockWasAdded: Bool) in
```

Comment:
remove this, a UITableViewCell, is just a dumb container that holds views, nothing else